### PR TITLE
Account-management UI Implementation

### LINKED ISSUE #41: Account Management
## Is your feature request related to a problem? Please describe.
Users need to be able to add and join houses.

## Describe the solution you'd like
Once a user is logged in, they need to be able to create or join a house. Detailed specification for the entire milestone is on [this page](https://github.com/701-T4/platform/wiki/Account-Management-Specification).

For this task, the specs are as follows (not yet final):
### Frontend
- New users will be shown a screen where they can either create or join a house
  - Joining a house will require a code
  - Creating a house will ask the user for some basic information, such as a name
- Users in a house will be directed to the dashboard for that house
  - This task doesn't require the dashboard to be implemented, just that they are redirected to the page

#### Screens
- Page to create new or join house
- Page to create a new flat

### Backend
- New users are created in the database with the Firebase (#40)
  - Please modify the guard to create a new user in the db if the auth is correct but user doesn't exist. See comments on issue #40 and related PR #51 for details
- Endpoint to get the user's current house
- Endpoint to create a new house, owned by the creator.
  - Usable by people who are not in a house
  - The house must have a unique associated code that can be used to join it
- Endpoint to join a house with a code, that will add the user to that house

### Checklist:

- [x] I have double checked that the issue does not currently exist. 
- [x] I have added the appropriate labels (e.g. feature, front-end, back-end).
- [x] I have added the **Platform Project** project

### PR DESCRIPTION
**Pull request by ywu666**
# Description

Implement the UI of the account-management page. The background colour will use the CSS code provided by the landing page. 

Fixes/resolves #41 

## Screenshots
image

image

After clicking on Create:

image

## Type of change
Please delete options that are not relevant.

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:
Leave blank if not applicable

I have completed these steps when making this pull request:
- [x] I have assigned my name to the issue
- [x] I have moved the issue to the **In Progress** column
- [x] I have labelled the PR appropriately
- [x] I have assigned myself to the PR

Before opening the PR for review:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have moved the linked issue to the **Review in Progress** column

### CODE REVIEW

**Comment by j-chad** [comment]
please don't add PRs to the projects and milestones 😃

File: packages/frontend/src/pages/public/AccountMgmtPage.tsx
Comment (Sparye) [inline packages/frontend/src/pages/public/AccountMgmtPage.tsx]: Is John a placeholder for now until the account backend is ready?
@frasermcc9 Do we merge this PR now? If so, do we need to create another issue to remind us come back to change it later?

**Review by Sparye — DISMISSED** [review]

File: packages/frontend/src/pages/public/AccountMgmtPage.tsx
Comment (frasermcc9) [reply packages/frontend/src/pages/public/AccountMgmtPage.tsx]: Should be on master now. `useAuth` should have it (or `getAuth().currentUser` as a fallback)

File: packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx]: can this be `/account` and `<ManageAccountPage />`

File: packages/frontend/src/styles/index.css
Comment (frasermcc9) [inline packages/frontend/src/styles/index.css:6]: Not sure how comfortable I am having every page being this gradient background, it's quite difficult to design for.

I think these starter pages are fine, but the main app should be white background (if light mode)

File: packages/frontend/src/styles/index.css
Comment (frasermcc9) [inline packages/frontend/src/styles/index.css:8]: Also for future reference, you can use tailwind in CSS files like this:
```css
body, html {
  @apply h-full bg-gradient-to-b from-teal-400 to-teal-600
}
```

File: packages/frontend/src/styles/index.css
Comment (FriedhelmWS) [inline packages/frontend/src/styles/index.css]: Are we able to refactor this linear-gradient colour for reuse?

**Review by FriedhelmWS — COMMENTED** [review]
Just a quiet note that in my opinion the row/col stuff kinda makes the entire structure messy as it makes a lot of tags, use of tailwind CSS is more concise in my opinion.

**Comment by frasermcc9** [comment]
> Just a quiet note that in my opinion the row/col stuff kinda makes the entire structure messy as it makes a lot of tags, use of tailwind CSS is more concise in my opinion.

Yeah i agree with this. With tailwind its basically just:
```tsx
<div className="flex flex-col lg:flex-row gap-8">
  <JoinButton />
  <AddButton />
</div>
```

File: packages/frontend/src/styles/index.css
Comment (frasermcc9) [reply packages/frontend/src/styles/index.css]: We should probably just use tailwind gradients if they look good, something like `bg-gradient-to-b from-teal-400 to-teal-600` (or whatever)

Comment (FriedhelmWS) [reply packages/frontend/src/styles/index.css]: yep because in the landing page which currently Johnny and I have been working on we just use the tailwind for the gradient.

File: packages/frontend/src/pages/public/AccountMgmtPage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/public/AccountMgmtPage.tsx]: OK. I will update soon.

File: packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx
Comment (ywu666) [reply packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx]: Ok

**Comment by ywu666** [comment]
Currently, I just use the components of NextUI. If you guys think tailwind it's better. I will change to the tailwind. For the background colour, cause @FriedhelmWS is taken on the landing page. I think I can reuse that background CSS code. I will delete this in my part. I think without using grid layout (row. col), then the responsive design is a problem

File: packages/frontend/src/styles/index.css
Comment (ywu666) [reply packages/frontend/src/styles/index.css]: I will change to the tailwind. btw, do most of our UI use tailwind instead NextUI?

File: packages/frontend/src/styles/index.css
Comment (ywu666) [reply packages/frontend/src/styles/index.css:6]: I think for other pages, the background colour can change to white (except the header part). Can we add CSS to div to achieve this (e.g. div will cover the background)? 
I am not sure where do I put this background CSS code will have no influence on the other pages.

File: packages/frontend/src/pages/public/AccountMgmtPage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/public/AccountMgmtPage.tsx]: image
My code is like this now to get the current user's display name. Please correct me if I am wrong. It seems to work but has a warning. I am not sure why this happens.
image

@Sparye @frasermcc9

File: packages/frontend/src/styles/index.css
Comment (j-chad) [reply packages/frontend/src/styles/index.css]: we should use nextui for most of our components if we can so we have a coherent design language

File: packages/frontend/src/pages/public/AccountMgmtPage.tsx
Comment (frasermcc9) [reply packages/frontend/src/pages/public/AccountMgmtPage.tsx]: It's probably because you aren't logged in, we'll need to fix the type definitions on that hook.

File: packages/frontend/src/styles/index.css
Comment (frasermcc9) [reply packages/frontend/src/styles/index.css:6]: You should be able to create a div on your page that takes up the full page `h-screen` and set the colour on that div.

File: packages/frontend/src/pages/public/AccountMgmtPage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/public/AccountMgmtPage.tsx]: OK. Then I will leave the code like this, and add some comments on it.

File: packages/frontend/src/pages/public/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/public/ManageAccountPage.tsx]:
```suggestion
  const { user } = useAuth();
```

File: packages/frontend/src/pages/public/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/public/ManageAccountPage.tsx]:
```suggestion
        text={user?.displayName}
```

**Comment by frasermcc9** [comment]
We'll have to wait for backend to finish to complete and merge this.

File: packages/frontend/src/pages/public/ManageAccountPage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/public/ManageAccountPage.tsx]: It cause the type error. image
image

Comment (frasermcc9) [reply packages/frontend/src/pages/public/ManageAccountPage.tsx]: `user?.displayName ?? ""`

File: packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx]: These routes are for users than aren't signed in, i.e.
```tsx
<Route path="auth" element={<SignInPage />} />
<Route path="*" element={<Navigate to="/home" replace />} />
```

All other pages should be placed in the `AuthenticatedRoutes` component since only signed in users should be able to access them.

File: packages/frontend/src/pages/public/CreateHousePage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/public/CreateHousePage.tsx]: This file, alongside `ManageAccountPage`, should be placed in a new `private` directory as they are only for users logged in.

File: packages/frontend/src/pages/public/CreateHousePage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/public/CreateHousePage.tsx]: Does this modal actually need to be its own route? Can it just be a component that's used in `ManageAccountPage`

i.e. just make this entire component a normal component instead of a page, and use it inside `ManageAccountPage`

File: packages/frontend/src/pages/public/ManageAccountPage.tsx
Comment (frasermcc9) [reply packages/frontend/src/pages/public/ManageAccountPage.tsx]: Alternatively, you can get the users profile picture. Something like
```suggestion
       icon={<img alt="Avatar" className="rounded-full" src={user?.photoURL ?? ''} />}
```
may be more appropriate.

**Review by frasermcc9 — CHANGES_REQUESTED** [review]
See comments on code

File: packages/frontend/src/pages/public/CreateHousePage.tsx
Comment (FriedhelmWS) [reply packages/frontend/src/pages/public/CreateHousePage.tsx]: Agree. It is nice to refactor this modal out so that could potentially reuse and make sure the style thing is consistent as modal may be needed to appear a lot across the entire website.

File: packages/frontend/src/pages/public/ManageAccountPage.tsx
Comment (FriedhelmWS) [inline packages/frontend/src/pages/public/ManageAccountPage.tsx]: Should we provide some technical standards for aria if we'd include them? Afaik aria is related to accessibility purpose and there is somehow related to screen reader etc.

File: packages/frontend/src/pages/public/ManageAccountPage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/public/ManageAccountPage.tsx]: Understand, so we can get the profile image from the google authentication. I will update it.

File: packages/frontend/src/pages/public/CreateHousePage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/public/CreateHousePage.tsx]: @FriedhelmWS  I agree with you. Initially, we put the modal inside the ManagementAccount Page. Cause in the issue, it mentioned that creating house be a page. So we move it as a page.  @uoa-aron527 will move this back.

Comment (uoa-aron527) [reply packages/frontend/src/pages/public/CreateHousePage.tsx]: I have moved it back to ManageAccountPage and the other changes are also made.

File: packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/routes/UnauthenticatedRoutes.tsx]: <Routes>
      <Route path="auth" element={<SignInPage />} />
      <Route path="landing" element={<LandingPage />} />
      <Route path="*" element={<Navigate to="/landing" replace />} />
    </Routes>

These are the only things that should be here. Everything else is only for users that are logged in.

File: packages/frontend/src/pages/routes/AuthenticatedRoutes.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/routes/AuthenticatedRoutes.tsx]: Move wildcard paths below all the others.

**Comment by frasermcc9** [comment]
Are there some files missing off the commits? Im getting import errors i.e.  
image

**Comment by ywu666** [comment]
Thank you for mentioning that, I have already updated it.

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: If a user is in a house they can't create or join a house. Instead simply navigate them to the dashboard.

```ts
const navigate = useNavigate()
navigate("/home")
```

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]:
```suggestion
```

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: Many of these should be reduced into a single `useState`, i.e.

```tsx
interface HouseDetails {
  phone: string;
  address: string;
  name: string;
}

const [houseDetails, setHouseDetails] = useState<HouseDetails>({phone: "", address: "", name: ""})
```

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: - Change to async/await
- Change then to await

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: Once the house has been joined, you should `navigate` the user to `/home`

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: Once a house has been created, you should `navigate` the user to `/home`

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: Is this your name, or the name for the house? Check with backend.

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx:13]: These are removable since either both are visible, or the user is redirected (either the user can create/join a flat, or they are already in one and should be navigated to `/home`)

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: Can this modal be moved into a separate component that is then used in this page?

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: Can this modal be moved into a separate component and then used on this page?

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: use `className="p-10"`

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: use `className="p-10"`

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: `className="absolute top-1/2 left-1/2 -translate-x-[10%] -translate-y-1/2"` over inline style

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [inline packages/frontend/src/pages/private/ManageAccountPage.tsx]: `className="float-right m-5"` over inline style

**Review by frasermcc9 — CHANGES_REQUESTED** [review]

**Comment by frasermcc9** [comment]
There's still other things that need addressing

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (ywu666) [reply packages/frontend/src/pages/private/ManageAccountPage.tsx]: Just want to confirm, Do you mean this?
image

**Comment by ywu666** [comment]
> There's still other things that need addressing

**Comment by ywu666** [comment]
Sorry, closed by accident. I finished the fixing.

**Comment by j-chad** [comment]
Are the screenshots up to date?

File: packages/frontend/src/components/CreateHouseModal.tsx
Comment (frasermcc9) [inline packages/frontend/src/components/CreateHouseModal.tsx]: Make a directory inside `components` for these. The intent is each page has a folder inside `components`, so for example, place these modals in `/components/account`

File: packages/frontend/src/pages/private/ManageAccountPage.tsx
Comment (frasermcc9) [reply packages/frontend/src/pages/private/ManageAccountPage.tsx]: Yes, but i dont think you need the state if you are navigating them.

File: packages/frontend/src/components/account/JoinHouseModal.tsx
Comment (frasermcc9) [inline packages/frontend/src/components/account/JoinHouseModal.tsx]: I dont think this hook is needed (houseData is never used, only the setter, so neither is probably needed)

File: packages/frontend/src/components/account/CreateHouseModal.tsx
Comment (frasermcc9) [inline packages/frontend/src/components/account/CreateHouseModal.tsx]: This response isn't needed. I think you just need to navigate to `/home` after the `await` (right now after creating nothing happens).

**Review by frasermcc9 — CHANGES_REQUESTED** [review]
Just a couple tiny more things, I think its merge ready one these are done!

File: packages/frontend/src/components/account/CreateHouseModal.tsx
Comment (frasermcc9) [inline packages/frontend/src/components/account/CreateHouseModal.tsx]: Last thing from me (promose)
```suggestion
            placeholder="Enter a name for your flat"
```

**Comment by ywu666** [comment]
hhha. no worries

File: packages/frontend/src/components/account/CreateHouseModal.tsx
Comment (frasermcc9) [inline packages/frontend/src/components/account/CreateHouseModal.tsx]:
```suggestion
            placeholder="Enter a name for your flat"
```

extra space 😂

**Review by frasermcc9 — APPROVED** [review]

**Comment by frasermcc9** [comment]
All seems to work for me now 😄

**Review by uoa-aron527 — APPROVED** [review]
Looks great! Good work :-)

**Review by wshe874 — APPROVED** [review]
LGTM

**Pull request merged**
